store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -409,6 +406,9 @@ private CarbonFileInputFormat prepareFileInputFormat(Job job, boolean enableBloc
     CarbonFileInputFormat format = prepareFileInputFormat(job, enableBlockletDistribution, false);
     List<InputSplit> splits =
         format.getSplits(new JobContextImpl(job.getConfiguration(), new JobID()));
+    // Clear the datamap cache as it is added in getSplits() method
```

Comment:
Can u add try finally here so In case of any exception it will clear the data map , pls check

Reply:
ok done.